parsing/lexer.mll

```diff
@@ -420,33 +469,52 @@ rule token = parse
           (Reserved_sequence (".~", Some "is reserved for use in MetaOCaml")) }
   | "~" raw_ident_escape (lowercase identchar * as name) ':'
       { LABEL name }
-  | "~" (lowercase identchar * as name) ':'
+  | "~" raw_ident_escape (ident_ext as raw_name) ':'
+      { let name = ident_for_extended lexbuf raw_name in
+        if UString.is_capitalized name
+        then error lexbuf (Capitalized_label name);
+        LABEL name }
+  | "~" (identstart identchar * as name) ':'
       { check_label_name lexbuf name;
         LABEL name }
-  | "~" (lowercase_latin1 identchar_latin1 * as name) ':'
-      { warn_latin1 lexbuf;
+  | "~" (ident_ext as raw_name) ':'
```

Comment:
There's now 4 very similar rules for labels ! Any way to collapse them to 1 or 2 ?